src/core/createReferenceObject.js

```diff
@@ -0,0 +1,36 @@
+const createBoundingBox = require('./createBoundingBox.js');
+
+//Create a reference object for an element
+module.exports.getRef= function (cyElement, userOptions) {
+    //Determine element properties to determine how to generate a reference object
```

Comment:
Similarly to a later comment, functions like these should be generic.  They could accept functions as arguments if you need customisation per type.